core/src/main/java/org/jdbi/v3/core/statement/SqlStatement.java

```diff
@@ -1794,8 +1794,12 @@ PreparedStatement internalExecute() {
         return stmt;
     }
 
-    PreparedStatement createStatement(final StatementContext ctx, ParsedSql parsedSql) throws SQLException {
-        return getHandle().getStatementBuilder().create(getHandle().getConnection(), parsedSql.getSql(), ctx);
+    PreparedStatement createStatement(final String parsedSql) throws SQLException {
```

Comment:
Can this be pushed up to `BaseStatement` so it can be shared with `Batch`?

Reply:
yes-ish but not a fan. BaseStatement then slowly becomes a collection of utility methods. We could push it to a helper.

Reply:
also means we would need to split the prepare/cleanup pair as the cleanup method needs the sql in the statement (that the base statement does not have).

Why do we need the sql on the close BTW?

Reply:
> Can this be pushed up to `BaseStatement` so it can be shared with `Batch`?

actually, batch does not have sql. So the create method there does not take a sql string.

Reply:
to sum this up: We *could* push all of this into the handle. Not sure that this is a good place, but it would become shorthand for getHandle().getStatementBuilder()

I think we should rethink the whole statementbuilder thing. it is actually handle specific (because that is where the connection lives) and the various sqlstatement pieces should just go there and say "stamp me a new Statement/PrepareStatement" from this statementcontext. And this should be very doable because we actually assign the parsed sql to the statement context (in SqlStatement#parseSql), so passing it in is actually not needed.

It is a pretty mess. I think I will not touch it on this PR but just clean up the uses and then we may consider a backwards incompatible changes of an API that most likely no one uses anyway. :-)